{"name": "Class Validator", "module": "java.xml", "package": "javax.xml.validation", "text": "A processor that checks an XML document against Schema.\n\n \n A validator object is not thread-safe and not reentrant.\n In other words, it is the application's responsibility to make\n sure that one Validator object is not used from\n more than one thread at any given time, and while the validate\n method is invoked, applications may not recursively call\n the validate method.", "codes": ["public abstract class Validator\nextends Object"], "fields": [], "methods": [{"method_name": "reset", "method_sig": "public abstract void reset()", "description": "Reset this Validator to its original configuration.\n\n Validator is reset to the same state as when it was created with\n Schema.newValidator().\n reset() is designed to allow the reuse of existing Validators\n thus saving resources associated with the creation of new Validators.\n\n The reset Validator is not guaranteed to have\n the same LSResourceResolver or ErrorHandler\nObjects, e.g. Object.equals(Object obj).\n It is guaranteed to have a functionally equal\n LSResourceResolver and ErrorHandler."}, {"method_name": "validate", "method_sig": "public void validate (Source source)\n              throws SAXException,\n                     IOException", "description": "Validates the specified input.\n\n This is just a convenience method for\n validate(Source source, Result result)\n with result of null."}, {"method_name": "validate", "method_sig": "public abstract void validate (Source source,\n                              Result result)\n                       throws SAXException,\n                              IOException", "description": "Validates the specified input and send the augmented validation\n result to the specified output.\n\n This method places the following restrictions on the types of\n the Source/Result accepted.\n\n \nSource / Result Accepted\n\n\n\nStreamSource\nSAXSource\nDOMSource\nStAXSource\n\n\n\n\nnull\nOK\nOK\nOK\nOK\n\n\nStreamResult\nOK\nIllegalArgumentException\nIllegalArgumentException\nIllegalArgumentException\n\n\nSAXResult\nIllegalArgumentException\nOK\nIllegalArgumentException\nIllegalArgumentException\n\n\nDOMResult\nIllegalArgumentException\nIllegalArgumentException\nOK\nIllegalArgumentException\n\n\nStAXResult\nIllegalArgumentException\nIllegalArgumentException\nIllegalArgumentException\nOK\n\n\n\nTo validate one Source into another kind of\n Result, use the identity transformer (see\n TransformerFactory.newTransformer()).\n\n Errors found during the validation is sent to the specified\n ErrorHandler.\n\n If a document is valid, or if a document contains some errors\n but none of them were fatal and the ErrorHandler didn't\n throw any exception, then the method returns normally."}, {"method_name": "setErrorHandler", "method_sig": "public abstract void setErrorHandler (ErrorHandler errorHandler)", "description": "Sets the ErrorHandler to receive errors encountered\n during the validate method invocation.\n\n \n Error handler can be used to customize the error handling process\n during a validation. When an ErrorHandler is set,\n errors found during the validation will be first sent\n to the ErrorHandler.\n\n \n The error handler can abort further validation immediately\n by throwing SAXException from the handler. Or for example\n it can print an error to the screen and try to continue the\n validation by returning normally from the ErrorHandler\n\n If any Throwable is thrown from an ErrorHandler,\n the caller of the validate method will be thrown\n the same Throwable object.\n\n \nValidator is not allowed to\n throw SAXException without first reporting it to\n ErrorHandler.\n\n \n When the ErrorHandler is null, the implementation will\n behave as if the following ErrorHandler is set:\n \n class DraconianErrorHandler implements ErrorHandler {\n     public void fatalError( SAXParseException e ) throws SAXException {\n         throw e;\n     }\n     public void error( SAXParseException e ) throws SAXException {\n         throw e;\n     }\n     public void warning( SAXParseException e ) throws SAXException {\n         // noop\n     }\n }\n \n\n When a new Validator object is created, initially\n this field is set to null."}, {"method_name": "getErrorHandler", "method_sig": "public abstract ErrorHandler getErrorHandler()", "description": "Gets the current ErrorHandler set to this Validator."}, {"method_name": "setResourceResolver", "method_sig": "public abstract void setResourceResolver (LSResourceResolver resourceResolver)", "description": "Sets the LSResourceResolver to customize\n resource resolution while in a validation episode.\n\n \nValidator uses a LSResourceResolver\n when it needs to locate external resources while a validation,\n although exactly what constitutes \"locating external resources\" is\n up to each schema language.\n\n \n When the LSResourceResolver is null, the implementation will\n behave as if the following LSResourceResolver is set:\n \n class DumbLSResourceResolver implements LSResourceResolver {\n     public LSInput resolveResource(\n         String publicId, String systemId, String baseURI) {\n\n         return null; // always return null\n     }\n }\n \n\n If a LSResourceResolver throws a RuntimeException\n  (or instances of its derived classes),\n then the Validator will abort the parsing and\n the caller of the validate method will receive\n the same RuntimeException.\n\n \n When a new Validator object is created, initially\n this field is set to null."}, {"method_name": "getResourceResolver", "method_sig": "public abstract LSResourceResolver getResourceResolver()", "description": "Gets the current LSResourceResolver set to this Validator."}, {"method_name": "getFeature", "method_sig": "public boolean getFeature (String name)\n                   throws SAXNotRecognizedException,\n                          SAXNotSupportedException", "description": "Look up the value of a feature flag.\n\n The feature name is any fully-qualified URI.  It is\n possible for a Validator to recognize a feature name but\n temporarily be unable to return its value.\n Some feature values may be available only in specific\n contexts, such as before, during, or after a validation.\n\n Implementors are free (and encouraged) to invent their own features,\n using names built on their own URIs."}, {"method_name": "setFeature", "method_sig": "public void setFeature (String name,\n                       boolean value)\n                throws SAXNotRecognizedException,\n                       SAXNotSupportedException", "description": "Set the value of a feature flag.\n\n \n Feature can be used to control the way a Validator\n parses schemas, although Validators are not required\n to recognize any specific feature names.\n\n The feature name is any fully-qualified URI.  It is\n possible for a Validator to expose a feature value but\n to be unable to change the current value.\n Some feature values may be immutable or mutable only\n in specific contexts, such as before, during, or after\n a validation."}, {"method_name": "setProperty", "method_sig": "public void setProperty (String name,\n                        Object object)\n                 throws SAXNotRecognizedException,\n                        SAXNotSupportedException", "description": "Set the value of a property.\n\n The property name is any fully-qualified URI.  It is\n possible for a Validator to recognize a property name but\n to be unable to change the current value.\n Some property values may be immutable or mutable only\n in specific contexts, such as before, during, or after\n a validation.\n\n \n All implementations that implement JAXP 1.5 or newer are required to\n support the XMLConstants.ACCESS_EXTERNAL_DTD and\n XMLConstants.ACCESS_EXTERNAL_SCHEMA properties.\n\n \n\nAccess to external DTDs in source or Schema file is restricted to\n      the protocols specified by the XMLConstants.ACCESS_EXTERNAL_DTD\n      property.  If access is denied during validation due to the restriction\n      of this property, SAXException will be thrown by the\n      validate(Source) method.\n\n      Access to external reference set by the schemaLocation attribute is\n      restricted to the protocols specified by the\n      XMLConstants.ACCESS_EXTERNAL_SCHEMA property.\n      If access is denied during validation due to the restriction of this property,\n      SAXException will be thrown by the\n      validate(Source) method.\n   \n"}, {"method_name": "getProperty", "method_sig": "public Object getProperty (String name)\n                   throws SAXNotRecognizedException,\n                          SAXNotSupportedException", "description": "Look up the value of a property.\n\n The property name is any fully-qualified URI.  It is\n possible for a Validator to recognize a property name but\n temporarily be unable to return its value.\n Some property values may be available only in specific\n contexts, such as before, during, or after a validation.\n\n Validators are not required to recognize any specific\n property names.\n\n Implementors are free (and encouraged) to invent their own properties,\n using names built on their own URIs."}]}